suite-common/metadata-types/src/metadataTypes.ts

```diff
@@ -139,30 +137,60 @@ export abstract class AbstractMetadataProvider {
     }
 }
 
-export interface AccountMetadata {
-    key: string; // legacy xpub format (btc-like coins) or account descriptor (other coins)
-    fileName: string; // file name in dropbox
-    aesKey: string; // asymmetric key for file encryption
+export interface AccountLabels {
     accountLabel?: MetadataItem;
     outputLabels: { [txid: string]: { [index: string]: MetadataItem } };
     addressLabels: { [address: string]: MetadataItem };
 }
 
+export interface WalletLabels {
+    walletLabel?: string;
+}
+
+export type Labels = AccountLabels | WalletLabels;
+
 export type DeviceMetadata =
     | {
           status: 'disabled' | 'cancelled'; // user rejects "Enable labeling" on device
       }
-    | {
+    | ({
           status: 'enabled';
-          key: string; // master key for all values (Device and Account)
-          fileName: string; // file name in dropbox
-          aesKey: string; // asymmetric key for file encryption
-          walletLabel?: string;
-      };
+      } & LabelableEntityKeysByVersion);
+
+type Data = Record<
+    LabelableEntityKeys['fileName'], // unique "id" for mapping with labelable entitties
+    Labels
+>;
+
+/**
+ * DataType dictates shape of data.
+ * in the future, it could be
+ * 'labels' | 'passwords' | 'contacts'...
+ */
+export type DataType = 'labels';
+
+/**
+ * Representation of provider data stored in reducer
+ * properties 'tokens' and 'type' are needed to recreate corresponding provider instance
+ * others may be used in UI
+ */
+export type MetadataProvider = {
+    type: MetadataProviderType;
+    user: string;
+    tokens: Tokens;
+    isCloud: boolean;
+    // decrypted content of data per provider
+    data: Data;
+    clientId: string;
+};
 
 export interface MetadataState {
```

Comment:
nit: I'd have it in the reducer, just like all the other state types

Reply:
someone moved this type to `metadataTypes` on purpose so I'd keep it here now. I guess some of these types might be required by native and if not now, it might happen in the future? not sure.